Content.Client/_DV/CosmicCult/UI/Monument/MonumentMenu.xaml.cs

```diff
@@ -0,0 +1,190 @@
+    private readonly IEnumerable<InfluencePrototype> _influencePrototypes;
+    private readonly ButtonGroup _glyphButtonGroup;
+    private ProtoId<GlyphPrototype> _selectedGlyphProtoId = string.Empty;
+    private HashSet<ProtoId<GlyphPrototype>> _unlockedGlyphProtoIds = [];
+    public Action<ProtoId<GlyphPrototype>>? OnSelectGlyphButtonPressed;
+    public Action? OnRemoveGlyphButtonPressed;
+
+    public Action<ProtoId<InfluencePrototype>>? OnGainButtonPressed;
+    private int _ccEntropyValue = 0;
+
+    public MonumentMenu()
+    {
+        RobustXamlLoader.Load(this);
+        IoCManager.InjectDependencies(this);
+        _sprite = _ent.System<SpriteSystem>();
+
+        _glyphPrototypes = _proto.EnumeratePrototypes<GlyphPrototype>().ToList();
+        _glyphPrototypes.Sort((x, y) =>
+            string.Compare(x.Name, y.Name, StringComparison.CurrentCultureIgnoreCase));
+        _influencePrototypes = _proto.EnumeratePrototypes<InfluencePrototype>();
+
+        _glyphButtonGroup = new ButtonGroup();
+
+        RemoveGlyphButton.OnPressed += _ => OnRemoveGlyphButtonPressed?.Invoke();
+        SelectGlyphButton.OnPressed += _ => OnSelectGlyphButtonPressed?.Invoke(_selectedGlyphProtoId);
+
+        _cfg.OnValueChanged(DCCVars.CosmicCultistEntropyValue, entropy =>
+        {
+            _ccEntropyValue = entropy;
+        },
+        invokeImmediately: true);
+    }
+
+    public void UpdateState(MonumentBuiState state)
+    {
+        _selectedGlyphProtoId = state.SelectedGlyph;
+        _unlockedGlyphProtoIds = state.UnlockedGlyphs;
+
+        CultProgressBar.BackgroundStyleBoxOverride = new StyleBoxFlat { BackgroundColor = new Color(15, 17, 30) };
+        CultProgressBar.ForegroundStyleBoxOverride = new StyleBoxFlat { BackgroundColor = new Color(91, 62, 124) };
+
+        SelectGlyphButton.StyleClasses.Add("ButtonColorPurpleAndCool");
+        RemoveGlyphButton.StyleClasses.Add("ButtonColorPurpleAndCool");
+
+        UpdateBar(state);
+        UpdateEntropy(state);
+        UpdateGlyphs();
+        UpdateInfluences(state);
+    }
+
+    // Update all the entropy fields
+    private void UpdateBar(MonumentBuiState state)
+    {
+        var percentComplete = ((float) state.CurrentProgress / (float) state.TargetProgress) * 100f; //too many parenthesis & probably unnecessary float casts but I'm not taking any chances
+
+        percentComplete = Math.Min(percentComplete, 100);
+
+        CultProgressBar.Value = percentComplete;
+
+        ProgressBarPercentage.Text = Loc.GetString("monument-interface-progress-bar", ("percentage", percentComplete.ToString("0")));
+    }
+
+    // Update all the entropy fields
+    private void UpdateEntropy(MonumentBuiState state)
+    {
+        var availableEntropy = "thinking emoji"; //if you see this, problem.
+        if (_ent.TryGetComponent<CosmicCultComponent>(_player.LocalEntity, out var cultComp))
+        {
+            availableEntropy = cultComp.EntropyBudget.ToString();
+        }
```

Comment:
Ideally this would log the error but I don't know off the top of my head how to do that, but a silly fallback message also seems bad

```suggestion
        int availableEntropy = 0;
        if (_ent.TryGetComponent<CosmicCultComponent>(_player.LocalEntity, out var cultComp))
        {
            availableEntropy = cultComp.EntropyBudget;
        }
```